.travis.yml

```diff
@@ -27,8 +27,5 @@ jobs:
       install:
         - docker-compose build
       script:
-        - docker-compose up
-        - sleep 15 # Wait for cluster to become ready, TODO: do not sleep
-        - HEADERS="$(curl -I -s http://localhost:3000/)"
-        - STATUS="$(echo $HEADERS | head -n1)"
-        - if [[ "$STATUS" != *"200 OK"* ]]; then echo "$HEADERS"; exit 1; fi
+        - docker-compose up -d
+        - while curl -Isf http://localhost:3000; do sleep 1; done
```

Comment:
Should we not add these lines after that?
```
        - HEADERS="$(curl -I -s http://localhost:3000/)"
        - STATUS="$(echo $HEADERS | head -n1)"
        - if [[ "$STATUS" != *"200 OK"* ]]; then echo "$HEADERS"; exit 1; fi
```

Reply:
(Sorry, I'm not yet accustomed to GitHub's "review" system)

Reply:
-f checks if the code is a failure or not. saving all the hacks.